_src/pattern-library/sass/components/_forms.scss

```diff
@@ -0,0 +1,455 @@
+        }
+
+        &:checked + .input-label:before {
+            background: $input-check-radio-background-checked;
+            box-shadow: inset 00 0 0 2px palette(grayscale, white);
+        }
+    }
+}
+
+// ----------------------------
+// #TEXTAREA
+// ----------------------------
+.input-textarea {
+    width: input-width(base);
+    height: input-height(base);
+
+    &:focus,
+    &:active,
+    &.is-active {
+        border-color: $input-border-default-focus;
+        color: $input-text-default-hover;
+    }
+
+    &.textarea-short {
+        height: input-height(short);
+    }
+
+    &.textarea-tall {
+        height: input-height(tall);
+    }
+
+    &.textarea-block {
+        width: 100%;
+    }
+}
+
+// ----------------------------
+// #SLIDERS
+// ----------------------------
+.input-slider,
+input[type="range"] {
+    width: input-width(base);
+    border: none;
+    border-radius: $input-progress-border-radius;
+    -webkit-appearance: none;
+
+    // Chrome
+    &::-webkit-slider-runnable-track {
+        @include transition(all timing(xx-fast) ease-in-out 0s);
+        @include transform-origin(center center);
+        width: 100%;
+        height: rem($baseline/2);
+        border-radius: $input-slider-border-default-radius;
+        border: none;
+        background: $input-slider-track-default;
+        cursor: pointer;
+    }
+
+    &::-webkit-slider-thumb {
+        @include size(rem($baseline/2));
+        @include transition(all timing(xx-fast) ease-in-out 0s);
+        -webkit-appearance: none;
+        border: none;
+        border-radius: 50%;
+        background: $input-slider-handle-default;
+    }
+
+    // Mozilla/Firefox
+    &::-moz-range-track {
+        @include transition(all timing(xx-fast) ease-in-out 0s);
+        @include transform-origin(center center);
+        width: 100%;
+        height: rem($baseline/2);
+        border-radius: $input-slider-border-default-radius;
+        border: none;
+        background: $input-slider-track-default;
+        cursor: pointer;
+    }
+
+    &::-moz-range-thumb {
+        @include size(rem($baseline/2));
+        @include transition(all timing(xx-fast) ease-in-out 0s);
+        border: none;
+        border-radius: 50%;
+        background: $input-slider-handle-default;
+    }
+
+    // IE
+    &::-ms-fill-upper { // after the thumb
+        background: $input-slider-track-default;
+    }
+
+    &::-ms-fill-lower { // before the thumb
+        background: $input-slider-track-default;
+    }
+
+    &::-ms-track { // notches
+        border: none;
+        color: transparent;
+    }
+
+    &::-ms-thumb {
+        @include size(rem($baseline/2));
+        @include transition(all timing(xx-fast) ease-in-out 0s);
+        border: none;
+        border-radius: $input-slider-border-default-radius;
+        background: $input-slider-handle-default;
+    }
+
+    // Hover states
+    &:hover,
+    &.is-hovered,
+    &:focus,
+    &.is-focused {
```

Comment:
Lots of stuff is repeated in here. Can you create a scoped extend and use it across these rules?